pkg/manifest/manifest.go

```diff
@@ -151,17 +156,78 @@ func (m *Manifest) Include(excludeIdentifier *string, includeTechPreview *bool,
 	return nil
 }
 
+// ManifestEnabled returns whether the manifest is enabled given the defined cluster capabilities.
+func (m *Manifest) ManifestEnabled(capabilities configv1.ClusterVersionCapabilitiesStatus) bool {
+	annotations := m.Obj.GetAnnotations()
+	if annotations == nil {
+		return true
+	}
+	enabled := true
+	caps := getManifestCapabilities(annotations)
+	for _, c := range caps {
+		_, isEnabled := getCapabilityProperties(c, capabilities)
+		if !isEnabled {
+			enabled = false
+			break
+		}
+	}
+	return enabled
+}
+
+// GetManifestCapabilities returns the manifest's capabilities.
+func (m *Manifest) GetManifestCapabilities() []configv1.ClusterVersionCapability {
+	annotations := m.Obj.GetAnnotations()
+	if annotations == nil {
+		return nil
+	}
+	return getManifestCapabilities(annotations)
+}
+
+func getManifestCapabilities(annotations map[string]string) []configv1.ClusterVersionCapability {
+	val, ok := annotations[CapabilityAnnotation]
+	if !ok {
+		return nil
+	}
+	caps := strings.Split(val, "+")
+	allCaps := make([]configv1.ClusterVersionCapability, len(caps))
+
+	for i, c := range caps {
+		allCaps[i] = configv1.ClusterVersionCapability(c)
+	}
+	return allCaps
+}
+
+func getCapabilityProperties(capability configv1.ClusterVersionCapability,
+	capabilities configv1.ClusterVersionCapabilitiesStatus) (isKnown, isEnabled bool) {
+
+	isKnown = false
+	isEnabled = false
+	for _, knownCapability := range capabilities.KnownCapabilities {
+		if capability == knownCapability {
+			isKnown = true
+			break
+		}
+	}
+	if !isKnown {
+		return false, false
+	}
+	for _, enabledCapability := range capabilities.EnabledCapabilities {
+		if capability == enabledCapability {
+			isEnabled = true
+			break
+		}
+
+	}
+	return true, isEnabled
+}
```

Comment:
I think we should break this function in to two different functions i.e. one for known capabilities and another for enabled capabilities.

Reply:
That way it will be easier to maintain

Reply:
Yeah, I agree. But this method can just go away if I get rid of `ManifestEnabled`.